src/ir/type-updating.h

```diff
@@ -434,13 +439,23 @@ class TypeMapper : public GlobalTypeRewriter {
     : GlobalTypeRewriter(wasm), mapping(mapping) {}
 
   void map() {
-    // Map the types of expressions (curr->type, etc.) to their merged
-    // types.
-    mapTypes(mapping);
-
     // Update the internals of types (struct fields, signatures, etc.) to
     // refer to the merged types.
-    update();
+    auto oldToNewTypes = rebuildTypes();
+
+    // Compose the given mapping from old types to old types with the new
```

Comment:
"old types to old types"?

Reply:
Yeah, the mapping the user provided. Will try to clarify.